pycloudlib/azure/cloud.py

```diff
@@ -782,6 +793,88 @@ def _check_for_network_interfaces(self):
+
+        Raises:
+            ValueError: the specified VM size was not found or vCPU capability is invalid.
+        """
+        resource_sku_list = self.compute_client.resource_skus.list(
+            filter=f"location eq '{self.location}'"
+        )
+
+        if not resource_sku_list:
+            raise ValueError(
+                f"No resource SKUs available for location: {self.location}"
+            )
+
+        try:
+            resource_sku: ResourceSku = next(
+                resource_sku
+                for resource_sku in resource_sku_list
+                if instance_type == resource_sku.name
+            )
+        except StopIteration as e:
+            raise ValueError(f"Unknown Azure VM size: {instance_type}.") from e
+
+        if not resource_sku.capabilities:
+            raise ValueError(
+                f"No capabilities found for Azure VM size: {instance_type}"
+            )
+
+        try:
+            vcpu_capability: ResourceSkuCapabilities = next(
+                capability
+                for capability in resource_sku.capabilities
+                if capability.name == "vCPUs"
+            )
+        except StopIteration as e:
+            raise ValueError(
+                f"No vCPUs capability found for Azure VM size: {instance_type}."
+            ) from e
+
+        if (
+            vcpu_capability.value is None
+            or not vcpu_capability.value.isdigit()
+        ):
+            raise ValueError(
+                f"Invalid vCPU capability value: {vcpu_capability.value} for {instance_type}"
+            )
+
+        return resource_sku.family, int(vcpu_capability.value)
+
+    def _check_compute_quota(self, instance_type: str):
+        """Check if there is sufficient Azure Compute quota for the deployment.
+
+        Raises:
+            QuotaLimitError: the vCPU requirement exceeds the region quota limit.
+            InsufficientQuotaError: the vCPU requirement exceeds the remaining quota.
+        """
+        family, vcpu = self._get_resource_sku_info(instance_type)
+
+        usage_list = self.compute_client.usage.list(self.location)
+        usage = next(
+            usage for usage in usage_list if usage.name.value == family
+        )
+
+        if usage.limit < vcpu:
+            raise QuotaLimitError(
+                f"Instance vCPU requirements exceed the quota limit for {instance_type} in {self.location}. "
+                f"{instance_type} requires {vcpu} vCPU(s). The region quota limit is {usage.limit}."
+            )
+
+        remaining_vcpu = usage.limit - usage.current_value
+        if remaining_vcpu < vcpu:
+            raise InsufficientQuotaError(
+                f"Insufficient Azure Compute quota: {vcpu} vCPU(s) required, {remaining_vcpu} available."
+            )
+
+    @exponential_backoff(
+        exceptions=(InsufficientQuotaError,),
+        retries=15,
+        max_time=10800,  # 10800 3 hours ceiling to cover random_jitter
+        jitter=True,
+    )
```

Comment:
let's nuke the backoff stuff. this belongs in consumers of pycloudlib, not in pycloudlib itself.

Reply:
also remove all custom backoff code. thanks!